### DISCUSSION TITLE
调用chat的AI接口，传入stream为false,还是调用流式数据，或者返回报错

### DISCUSSION THREAD

**Discussion by yi-boide** [lobehub/lobehub#2523] [Development Question]
### 💻 Operating System

macOS

### 📦 Environment

Other

### 🌐 Browser

Chrome

### 🐛 Bug Description

调用chat的AI接口，传入stream为false,还是调用流式数据，或者返回报错，以及返回控制，无法处理的的数据
希望能够传入stream为false时，不要以流的方式返回数据，而是直接返回，例如在调用插件的时候应该是直接获取而不应该是流的形式获取，这样会更方便处理，以及提升一些响应时间

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
> 例如在调用插件的时候应该是直接获取而不应该是流的形式获取

这个不对的。现在已经支持了插件的流式处理，为了更好的用户体验，我建议你应该参考 OpenAI 的官方实现方式，以流式的方式传输插件内容。

**Reply by yi-boide** [reply]
可是获取插件信息的时候，以流式返回数据，用户是感知不到的，也是不关心的，只关心最终结果，返回文本的时候以流式返回，用户的确是能够直观看到结果的，我在想获取插件数据的时候，若是直接返回是不是更快一些，毕竟流的形式他需要一段一段推送，不过若是流式请求和不使用流式请求所花费的时间是一样的话拿到不用考虑这个问题
